services/mfa/mfa_test.go

```diff
@@ -72,24 +92,168 @@ func TestGetIssuerFromUrl(t *testing.T) {
 
 func TestActivate(t *testing.T) {
 	user := &model.User{Id: model.NewId(), Roles: "system_user"}
+	user.MfaSecret = b32.StdEncoding.EncodeToString([]byte(model.NewRandomString(MFA_SECRET_SIZE)))
+	token := dgoogauth.ComputeCode(user.MfaSecret, time.Now().UTC().Unix()/30)
 
 	config := model.Config{}
 	config.SetDefaults()
 	config.ServiceSettings.EnableMultifactorAuthentication = model.NewBool(true)
 	configService := testutils.StaticConfigService{Cfg: &config}
-	storeMock := mocks.Store{}
-	userStoreMock := mocks.UserStore{}
-	userStoreMock.On("UpdateMfaActive", user.Id, mock.AnythingOfType("bool")).Return(func(userId string, active bool) *model.AppError {
-		return nil
+
+	t.Run("fail on disabled mfa", func(t *testing.T) {
+		wrongConfig := model.Config{}
+		wrongConfig.SetDefaults()
+		wrongConfig.ServiceSettings.EnableMultifactorAuthentication = model.NewBool(false)
+		wrongConfigService := testutils.StaticConfigService{Cfg: &wrongConfig}
+		mfa := New(wrongConfigService, nil)
+		err := mfa.Activate(user, "not-important")
+		require.NotNil(t, err)
 	})
-	storeMock.On("User").Return(&userStoreMock)
 
-	mfa := Mfa{configService, &storeMock}
+	t.Run("fail on wrongly formatted token", func(t *testing.T) {
+		mfa := New(configService, nil)
+		err := mfa.Activate(user, "invalid-token")
+		require.NotNil(t, err)
```

Comment:
Does this test differentiate from the other one in the error returned ? Specifically I would want to test for `mfa.activate.authenticate.app_error` and `mfa.activate.bad_token.app_error`.